src/Arcus.Scripting.Management/Scripts/Remove-AzApiManagementSoftDeletedService.ps1

```diff
@@ -0,0 +1,38 @@
+Param(
+    [Parameter(Mandatory = $true)][string] $Name = $(throw "Name of the API Management instance is required"),
+    [Parameter(Mandatory = $false)][string] $SubscriptionId = "",
+    [Parameter(Mandatory = $false)][string] $AccessToken = ""
+)
+
+if($SubscriptionId -eq "" -or $AccessToken -eq ""){
+    # Request accessToken in case the script contains records
+    $token = Get-AzCachedAccessToken
```

Comment:
Great stuff in re-using existing functionality. implementation here looks real nice and clean! 💯

Reply:
Thanks for your suggestion on this 👍🏻